ui/app/AppLayouts/Chat/controls/community/CommunityOptions.qml

```diff
@@ -0,0 +1,77 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+ColumnLayout {
+    id: root
+
+    property alias archiveSupportOptionVisible: archiveSupport.visible
+
+    property alias archiveSupportEnabled: archiveSupportToggle.checked
+    property alias requestToJoinEnabled: requestToJoinToggle.checked
+    property alias pinMessagesEnabled: pinMessagesToggle.checked
+
+    spacing: 0
+
+    QtObject {
+        id: d
+        readonly property real optionHeight: 64
+    }
+
+    RowLayout {
+        id: archiveSupport
+
+        Layout.fillWidth: true
+
+        StatusBaseText {
+            text: qsTr("Community history service")
+        }
+
+        Item {
+            Layout.fillWidth: true
+            implicitHeight: d.optionHeight
+        }
+
+        StatusCheckBox {
+            id: archiveSupportToggle
+        }
+    }
```

Comment:
I did it [initially](https://github.com/status-im/status-desktop/blob/cce788e4340a43d89eb95b6057d318b1df082bac/ui/app/AppLayouts/Chat/controls/community/CommunityOptions.qml) that way, but it's against the Design.

I would love to use an inline component, the feature introduced in Qt 5.15. It would remove some duplication from here, but since we are 5.14 I decided to live with duplication :sweat: